storm-core/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -1382,5 +1422,438 @@ public static TopologyInfo getTopologyInfo(String name, String asUser, Map storm
+    }
+
+    protected void forceDeleteImpl(String path) {
+        LOG.debug("Deleting path {}", path);
+        if (checkFileExists(path)) {
+            try {
+                FileUtils.forceDelete(new File(path));
+            } catch (IOException ignored) {}
+        }
+    }
+
+    /**
+     * Creates a symbolic link to the target
+     * @param dir the parent directory of the link
+     * @param targetDir the parent directory of the link's target
+     * @param targetFilename the file name of the links target
+     * @param filename the file name of the link
+     * @return the path of the link if it did not exist, otherwise null
+     * @throws IOException
+     */
+    public static Path createSymlink(String dir, String targetDir,
+            String targetFilename, String filename) throws IOException {
+        Path path = Paths.get(dir, filename).toAbsolutePath();
+        Path target = Paths.get(targetDir, targetFilename).toAbsolutePath();
+        LOG.debug("Creating symlink [{}] to [{}]", path, target);
+        if (!path.toFile().exists()) {
+            return Files.createSymbolicLink(path, target);
+        }
+        return null;
+    }
+
+    /**
+     * Convenience method for the case when the link's file name should be the
+     * same as the file name of the target
+     */
+    public static Path createSymlink(String dir, String targetDir,
+                                     String targetFilename) throws IOException {
+        return Utils.createSymlink(dir, targetDir, targetFilename,
+                targetFilename);
+    }
+
+    /**
+     * Returns a Collection of file names found under the given directory.
+     * @param dir a directory
+     * @return the Collection of file names
+     */
+    public static Collection<String> readDirContents(String dir) {
+        Collection<String> ret = new HashSet<>();
+        File[] files = new File(dir).listFiles();
+        if (files != null) {
+            for (File f: files) {
+                ret.add(f.getName());
+            }
+        }
+        return ret;
+    }
+
+    /**
+     * Returns the value of java.class.path System property. Kept separate for
+     * testing.
+     * @return the classpath
+     */
+    public static String currentClasspath() {
+        return _instance.currentClasspathImpl();
+    }
+
+    public String currentClasspathImpl() {
+        return System.getProperty("java.class.path");
+    }
+
+    /**
+     * Returns a collection of jar file names found under the given directory.
+     * @param dir the directory to search
+     * @return the jar file names
+     */
+    private static Collection<String> getFullJars(String dir) {
+        File[] files = new File(dir).listFiles(new FilenameFilter() {
+            @Override
+            public boolean accept(File dir, String name) {
+                return name.endsWith(".jar");
+            }
+        });
+        Collection<String> ret = new HashSet<String>(files.length);
```

Comment:
We need to check for files if it is null or not.  Previously if a dir did not exist getFullJars would return an empty set.  Now it will NPE here.